pkg/network/common/common.go

```diff
@@ -73,6 +74,15 @@ func ParseClusterNetwork(cn *networkv1.ClusterNetwork) (*ParsedClusterNetwork, e
 		utilruntime.HandleError(fmt.Errorf("Configured serviceNetworkCIDR value %q is invalid; treating it as %q", cn.ServiceNetwork, pcn.ServiceNetwork.String()))
 	}
 
+	pcn.MachineNetwork, err = networkutils.ParseCIDRMask(cn.Network)
```

Comment:
`cn.Network` isn't the machine network, it's the deprecated single-value form of `cn.ClusterNetworks`.
There is no CIDR in an OCP cluster that reliably includes the nodes and nothing else.

Reply:
I didn't want to bother you on weekends but since you are here, I think that will be easier to send pod traffic through the tunnel and avoid the asymetric traffic 

Pods traffic to nodes should not be masquerade, however, this creates an asymmetric routing problem ... should traffic directed to nodeIPs directed through the tunnel interface?

Reply:
> Pods traffic to nodes should not be masquerade

Says who?

Reply:
https://kubernetes.io/docs/concepts/cluster-administration/networking/

> Kubernetes imposes the following fundamental requirements on any networking implementation (barring any intentional network segmentation policies):
> 
> pods on a node can communicate with all pods on all nodes without NAT
> agents on a node (e.g. system daemons, kubelet) can communicate with all pods on that node
> Note: For those platforms that support Pods running in the host network (e.g. Linux):
> 
> pods in the host network of a node can communicate with all pods on all nodes without NAT

` pods in the host network of a node can communicate with all pods on all nodes without NAT`

Reply:
> ` pods in the host network of a node can communicate with all pods on all nodes without NAT`

That doesn't say that the reverse has to be true though.

OTOH

> pods on a node can communicate with all pods on all nodes without NAT

If you assume that the second "pods" includes host-network pods, then we'd be wrong.

But if you assume that the second "pods" includes host-network pods, then you should presumably also assume that the first "pods" includes host-network pods, in which case there is no need for the "pods in the host network ..." statement because it would be redundant. So that implies that "pods" alone is supposed to be understood to mean "pod-network pods".

Or maybe there's a difference between "pods" and "all pods"? ...

We should probably get this text clarified, and then make sure that there are conformance tests ensuring all of the actually-required behaviors.

Reply:
At any rate, if we want to cause pod-to-node traffic to be not-masqueraded, then that has to be done with per-node-ip rules, not with a CIDR match. The easiest way to do this would be in ovscontroller.go rather than iptables.go; have `AddHostSubnetRules` add VXLAN rules for `nw_dst=${subnet.HostIP}` too, not just `nw_dst=${subnet.Subnet}`